src/main/java/frc/robot/subsystems/BatterySubsystem.java

```diff
@@ -0,0 +1,221 @@
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj.simulation.DriverStationSim;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Robot;
+import java.io.File;
+import java.util.Map;
+import javax.sound.sampled.AudioInputStream;
+import javax.sound.sampled.AudioSystem;
+import javax.sound.sampled.Clip;
+import javax.swing.JOptionPane;
+import edu.wpi.first.wpilibj.simulation.RoboRioSim;
+
+/** Add your docs here. */
+// Put methods for controlling this subsystem
+// here. Call these from Commands.
+public class BatterySubsystem extends SubsystemBase {
+
+  private ShuffleboardTab btTab;
+  private NetworkTableEntry sbVoltage;
+  private NetworkTableEntry sbSimVoltage;
+  private NetworkTableEntry sbTimer;
+  private NetworkTableEntry sbTimerChange;
+  private NetworkTableEntry sbTimerHighCurrent;
+  // private DriverStation driverStation;
+  private Timer timer = new Timer();
+  private Timer currentTimer = new Timer();
+  // public DriverStationSim driverSim;
+  // Sim variables
+  private double simVolt;
+  private double simCurrent;
+  private boolean colorOn; // Variable for color change testing
+  // private int minutesPassed;
+
+  public BatterySubsystem() {
+    // init();
+    this.resetTimers();
+    btTab = Shuffleboard.getTab("Battery");
+    // Change 16 to desired value
+    Shuffleboard.getTab("Battery")
+        .addBoolean("Battery Voltage", () -> getVoltage() > Constants.minVoltageRed)
+        .withProperties(Map.of("colorWhenTrue", "red"))
+        .withProperties(Map.of("colorWhenFalse", "red"));
+    sbVoltage =
+        btTab.add("Current Battery Voltage", 0).withSize(2, 2).withPosition(0, 0).getEntry();
+    sbSimVoltage = btTab.add("Simulation Voltage", 0).withSize(2, 2).withPosition(2, 0).getEntry();
+    sbTimer = btTab.add("Timer", 0).withSize(2, 2).withPosition(0, 2).getEntry();
+    sbTimerHighCurrent =
+        btTab.add("High Current Timer", 0).withSize(2, 2).withPosition(0, 4).getEntry();
+    sbTimerChange = btTab.add("Change Timer", 0).withSize(2, 2).withPosition(4, 0).getEntry();
+    timer.start();
+    currentTimer.start();
+    simVolt = 16.0;
+    DriverStationSim.setSendError(true);
+    // btTab.addBoolean("Change the battery: ", valueSupplier)
+  }
+
+  public void periodic() {
+    sbVoltage.setDouble(getVoltage());
+    // sbInCurrent.setDouble(getInputCurrent());
+    sbSimVoltage.setNumber(RoboRioSim.getVInVoltage()); // Just temporary, replace with line above
+    sbTimer.setDouble(timer.get());
+    sbTimerHighCurrent.setDouble(currentTimer.get());
+    sbTimerChange.setBoolean(checkTimer()); // Replace checkVoltage() with checkTimer() if necessary
+    checkCurrent();
+    checkVoltage();
+    // checkMinute();
+  }
+
+  public double getVoltage() {
+    if (Robot.isSimulation()) {
+      return (int) simVolt;
+    } else {
+      return RobotController.getBatteryVoltage();
```

Comment:
I don't think you need a different return value for simulation. Since we are using the RoboRioSim to set the voltage it should get returned properly here.

Reply:
done, removed the variable

Reply:
changed it to return with a RoboRioSim.getVInVoltage() method, should I just use the getBatteryVoltage() one instead?